doc/ug/rust_for_c.md

```diff
@@ -0,0 +1,2302 @@
+
+
+#### Aliases {#aliases}
+
+Rust has `type`, which works exactly like `typedef` in `C`.
+Its syntax is
+```rust
+type MyAlias = u32;
+```
+
+### Expressions and Statements {#expressions-and-statements}
+
+Very much unlike C, Rust has virtually no statements in its grammar: almost everything is an expression of some kind, and can be used in expression context.
+Roughly speaking, the only statement in the language is creating a binding:
+```rust
+let x: u32 = foo();
+```
+The type after the `:` is optional, and if missing, the compiler will use all information in the current scope, both before and after the `let`, to infer one[^48].
+
+An expression followed by a semicolon simply evaluates the expression for side-effects, like in other languages[^49].
+Some expressions, like `if`s, `whiles`, and `for`s, don't need to be followed by a semicolon.
+If they are not used in an expression, they will be executed for side-effects.
+
+`let` bindings are immutable by default, but `let mut x = ...;` will make it mutable.
+
+Like in C, reassignment is an expression, but unlike in C, it evaluates to `()` rather than to the assigned value.
+
+Like in almost all other languages, literals, operators, function calls, variable references, and so on are all standard expressions, which we've already seen how Rust spells already.
+Let's dive into some of Rust's other expressions.
+
+
+#### Blocks {#blocks}
+
+Blocks in Rust are like better versions of C's blocks; in Rust, every block is an expression.
+A block is delimited by `{ }`, consists of a set of a list of statements and items, and potentially an ending expression, much like a function.
+The block will then evaluate to the final expression. For example:
+```rust
+let foo = {
+  let bar = 5;
+  bar ^ 2
+};
+```
+Blocks are like local functions that execute immediately, and can be useful for constraining the scope of variables.
+
+
+#### Conditionals: `if` and `match` {#conditionals-if-and-match}
+
+Rust's `if` expression is, syntactically, similar to C's. The full syntax is
+```rust
+if cond1 {
+  // ...
+} else if cond2 {
+  // ...
+} else {
+  // ...
+}
+```
+
+Conditions must be expressions that evaluate to `bool`.
+A conditional may have zero or more `else if` clauses, and the `else` clause is optional.
+Because of this, Rust does not need (and, thus, does not have) a ternary operator:
+```rust
+let x = if c { a } else { b };
+```
+
+Braces are required on `if` expressions in Rust.
+
+`if` expressions evaluate to the value of the block that winds up getting executed.
+As a consequence, all blocks must have the same type[^50].
+For example, the following won't compile, because of an errant semicolon:
+```rust
+if cond() {
+  my_int(4)   // Type is i32.
+} else {
+  my_int(7);  // Type is (), due to the ;
+}
+```
+
+In general, it is a good idea to end all expressions with a semicolon, unless its value is needed.
```

Comment:
Hmm... I'm not sure. The point I wanted to make was "when you use a conditional, all the values get unified; to avoid unification errors, remember to include a semicolon". I think this advice applies less strongly to blocks that are actively being used as expressions, where the semicolon is actually totally unimportant unless you assign the block to a `let`.

Maybe it should be reworded though, since the semicolons are _required_ for expressions prior to the final one?

Reply:
If I understand you, your contention is that this is the main place the final semicolon (or not) will bite you, and that the end of a function's block is less likely because of explicit `return`. 

Ok, I can get behind that. Maybe do a forward reference to `potential ending expression` from `Blocks` to this paragraph instead then?

Reply:
Done.